Seeding not triggered when the DB is reset via migrate dev

## Bug description

Seeding works by calling `prisma migrate reset`, but not when calling `prisma migrate dev` and the database is reset interactively.


## Expected behavior

Seeding is triggered when the DB is reset

## Debug output

```
> DEBUG=* DATABASE_URL="mysql://prismauser:prismauser@localhost/prismatest" npx prisma migrate dev  --preview-feature
  tryLoadEnv Environment variables not found at null +0ms
  tryLoadEnv Environment variables loaded from ./prisma/.env +1ms
Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": MySQL database "prismatest" at "localhost:undefined"

warn "nativeTypes" is a preview feature. There may be bugs and it's not recommended to use it in production environments.
  MigrateEngine:rpc starting migration engine with binary: /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/engines/migration-engine-darwin +0ms
  MigrateEngine:rpc SENDING RPC CALL {"id":1,"jsonrpc":"2.0","method":"devDiagnostic","params":{"migrationsDirectoryPath":"/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/migrations"}} +3ms
  MigrateEngine:stderr Jan 19 13:06:39.632  INFO migration_engine: Starting migration engine RPC server git_hash="94e94e327ccbd6d374c9a079ed59f89b077348e4" +0ms
  MigrateEngine:stderr Jan 19 13:06:39.637  INFO quaint::single: Starting a mysql connection. +4ms
  MigrateEngine:stderr Jan 19 13:06:39.646  INFO DevDiagnostic:calculate_drift:sql_schema_from_migration_history: quaint::single: Starting a mysql connection. +9ms
  MigrateEngine:stderr Jan 19 13:06:39.857  INFO DevDiagnostic:validate_migrations:sql_schema_from_migration_history: quaint::single: Starting a mysql connection. +212ms
  migrate:dev {
  migrate:dev   devDiagnostic: '{\n' +
  migrate:dev     '  "action": {\n' +
  migrate:dev     '    "tag": "reset",\n' +
  migrate:dev     '    "reason": "The following migration(s) are applied to the database but missing from the local migrations directory: 20210119130328_dontknow"\n' +
  migrate:dev     '  }\n' +
  migrate:dev     '}'
  migrate:dev } +0ms
✔ The following migration(s) are applied to the database but missing from the local migrations directory: 20210119130328_dontknow

We need to reset the MySQL database "prismatest" at "localhost:undefined".
Do you want to continue? All data will be lost. … yes

  MigrateEngine:rpc SENDING RPC CALL {"id":2,"jsonrpc":"2.0","method":"reset","params":{}} +3s
  MigrateEngine:rpc SENDING RPC CALL {"id":3,"jsonrpc":"2.0","method":"applyMigrations","params":{"migrationsDirectoryPath":"/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/migrations"}} +12ms
  MigrateEngine:stderr Jan 19 13:06:42.528  INFO ApplyMigrations:Applying migration{migration_name="20210119114009_init"}: migration_core::commands::apply_migrations: Applying `20210119114009_init` script="-- CreateTable\nCREATE TABLE `Post` (\n    `id` INT NOT NULL AUTO_INCREMENT,\n    `title` VARCHAR(191) NOT NULL,\n    `content` VARCHAR(191),\n    `published` BOOLEAN NOT NULL DEFAULT false,\n    `authorId` INT,\n\n    PRIMARY KEY (`id`)\n) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;\n\n-- CreateTable\nCREATE TABLE `User` (\n    `id` INT NOT NULL AUTO_INCREMENT,\n    `email` VARCHAR(191) NOT NULL,\n    `name` VARCHAR(191),\nUNIQUE INDEX `User.email_unique`(`email`),\n\n    PRIMARY KEY (`id`)\n) DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci;\n\n-- AddForeignKey\nALTER TABLE `Post` ADD FOREIGN KEY (`authorId`) REFERENCES `User`(`id`) ON DELETE SET NULL ON UPDATE CASCADE;\n" +3s
  MigrateEngine:stderr Jan 19 13:06:42.631  INFO ApplyMigrations:Applying migration{migration_name="20210119114525_add_some_changes_native_types"}: migration_core::commands::apply_migrations: Applying `20210119114525_add_some_changes_native_types` script="-- DropForeignKey\nALTER TABLE `Post` DROP FOREIGN KEY `post_ibfk_1`;\n\n-- AlterTable\nALTER TABLE `User` ADD COLUMN     `firstName` VARCHAR(100),\n    ADD COLUMN     `lastName` VARCHAR(300),\n    MODIFY `name` VARCHAR(500);\n\n-- AddForeignKey\nALTER TABLE `Post` ADD FOREIGN KEY (`authorId`) REFERENCES `User`(`id`) ON DELETE SET NULL ON UPDATE CASCADE;\n" +103ms
The following migration(s) have been applied:

migrations/
  └─ 20210119114009_init/
    └─ migration.sql
  └─ 20210119114525_add_some_changes_native_types/
    └─ migration.sql
  MigrateEngine:rpc SENDING RPC CALL {"id":4,"jsonrpc":"2.0","method":"evaluateDataLoss","params":{"migrationsDirectoryPath":"/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/migrations","prismaSchema":"generator client {\n  provider        = \"prisma-client-js\"\n  previewFeatures = [\"nativeTypes\"]\n}\n\ndatasource db {\n  provider = \"mysql\"\n  url      = env(\"DATABASE_URL\")\n}\n\nmodel Post {\n  id        Int     @id @default(autoincrement()) @db.Int\n  title     String  @db.VarChar(191)\n  content   String? @db.VarChar(191)\n  published Boolean @default(false) @db.TinyInt\n  authorId  Int?    @db.Int\n  author    User?   @relation(fields: [authorId], references: [id])\n\n  @@index([authorId], name: \"authorId\")\n}\n\nmodel User {\n  id    \tInt     @id @default(autoincrement()) @db.Int\n  email \tString  @unique @db.VarChar(191)\n  name  \tString? @db.VarChar(500)\n  firstName String? @db.VarChar(100)\n  lastName  String? @db.VarChar(300)\n  posts Post[]\n}\n"}} +223ms
  MigrateEngine:stderr Jan 19 13:06:42.727  INFO EvaluateDataLoss:infer_next_migration:sql_schema_from_migration_history: quaint::single: Starting a mysql connection. +96ms
  migrate:dev {
  migrate:dev   evaluateDataLossResult: {
  migrate:dev     migrationSteps: [
  migrate:dev       'ALTER TABLE `Post` DROP FOREIGN KEY `post_ibfk_1`',
  migrate:dev       'ALTER TABLE `Post` ADD FOREIGN KEY (`authorId`) REFERENCES `User`(`id`) ON DELETE SET NULL ON UPDATE CASCADE'
  migrate:dev     ],
  migrate:dev     warnings: [],
  migrate:dev     unexecutableSteps: []
  migrate:dev   }
  migrate:dev } +3s
✔ Name of migration … random
  MigrateEngine:rpc SENDING RPC CALL {"id":5,"jsonrpc":"2.0","method":"createMigration","params":{"migrationsDirectoryPath":"/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/migrations","migrationName":"random","draft":false,"prismaSchema":"generator client {\n  provider        = \"prisma-client-js\"\n  previewFeatures = [\"nativeTypes\"]\n}\n\ndatasource db {\n  provider = \"mysql\"\n  url      = env(\"DATABASE_URL\")\n}\n\nmodel Post {\n  id        Int     @id @default(autoincrement()) @db.Int\n  title     String  @db.VarChar(191)\n  content   String? @db.VarChar(191)\n  published Boolean @default(false) @db.TinyInt\n  authorId  Int?    @db.Int\n  author    User?   @relation(fields: [authorId], references: [id])\n\n  @@index([authorId], name: \"authorId\")\n}\n\nmodel User {\n  id    \tInt     @id @default(autoincrement()) @db.Int\n  email \tString  @unique @db.VarChar(191)\n  name  \tString? @db.VarChar(500)\n  firstName String? @db.VarChar(100)\n  lastName  String? @db.VarChar(300)\n  posts Post[]\n}\n"}} +7s
  MigrateEngine:stderr Jan 19 13:06:49.283  INFO CreateMigration{migration_name="random" draft=false}:infer_next_migration:sql_schema_from_migration_history: quaint::single: Starting a mysql connection. +7s
  migrate:dev {
  migrate:dev   createMigrationResult: { generatedMigrationName: '20210119130649_random' }
  migrate:dev } +7s
  MigrateEngine:rpc SENDING RPC CALL {"id":6,"jsonrpc":"2.0","method":"applyMigrations","params":{"migrationsDirectoryPath":"/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/migrations"}} +227ms
  MigrateEngine:stderr Jan 19 13:06:49.513  INFO ApplyMigrations:Applying migration{migration_name="20210119130649_random"}: migration_core::commands::apply_migrations: Applying `20210119130649_random` script="-- DropForeignKey\nALTER TABLE `Post` DROP FOREIGN KEY `post_ibfk_1`;\n\n-- AddForeignKey\nALTER TABLE `Post` ADD FOREIGN KEY (`authorId`) REFERENCES `User`(`id`) ON DELETE SET NULL ON UPDATE CASCADE;\n" +230ms

The following migration(s) have been created and applied from new schema changes:

migrations/
  └─ 20210119130649_random/
    └─ migration.sql

Running generate... (Use --skip-generate to skip the generators)
{
  prismaClientDir: '/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client'
}
  getGenerators Prisma Client +0ms
  getGenerators {
  getGenerators   datamodel: 'generator client {\n' +
  getGenerators     '  provider        = "prisma-client-js"\n' +
  getGenerators     '  previewFeatures = ["nativeTypes"]\n' +
  getGenerators     '}\n' +
  getGenerators     '\n' +
  getGenerators     'datasource db {\n' +
  getGenerators     '  provider = "mysql"\n' +
  getGenerators     '  url      = env("DATABASE_URL")\n' +
  getGenerators     '}\n' +
  getGenerators     '\n' +
  getGenerators     'model Post {\n' +
  getGenerators     '  id        Int     @id @default(autoincrement()) @db.Int\n' +
  getGenerators     '  title     String  @db.VarChar(191)\n' +
  getGenerators     '  content   String? @db.VarChar(191)\n' +
  getGenerators     '  published Boolean @default(false) @db.TinyInt\n' +
  getGenerators     '  authorId  Int?    @db.Int\n' +
  getGenerators     '  author    User?   @relation(fields: [authorId], references: [id])\n' +
  getGenerators     '\n' +
  getGenerators     '  @@index([authorId], name: "authorId")\n' +
  getGenerators     '}\n' +
  getGenerators     '\n' +
  getGenerators     'model User {\n' +
  getGenerators     '  id    \tInt     @id @default(autoincrement()) @db.Int\n' +
  getGenerators     '  email \tString  @unique @db.VarChar(191)\n' +
  getGenerators     '  name  \tString? @db.VarChar(500)\n' +
  getGenerators     '  firstName String? @db.VarChar(100)\n' +
  getGenerators     '  lastName  String? @db.VarChar(300)\n' +
  getGenerators     '  posts Post[]\n' +
  getGenerators     '}\n',
  getGenerators   datasources: [
  getGenerators     {
  getGenerators       name: 'db',
  getGenerators       provider: [Array],
  getGenerators       activeProvider: 'mysql',
  getGenerators       url: [Object]
  getGenerators     }
  getGenerators   ],
  getGenerators   generator: {
  getGenerators     name: 'client',
  getGenerators     provider: 'prisma-client-js',
  getGenerators     output: '/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client',
  getGenerators     binaryTargets: [],
  getGenerators     previewFeatures: [ 'nativeTypes' ],
  getGenerators     config: {}
  getGenerators   },
  getGenerators   dmmf: {
  getGenerators     datamodel: { enums: [], models: [Array] },
  getGenerators     schema: {
  getGenerators       inputObjectTypes: [Object],
  getGenerators       outputObjectTypes: [Object],
  getGenerators       enumTypes: [Object]
  getGenerators     },
  getGenerators     mappings: { modelOperations: [Array], otherOperations: [Object] }
  getGenerators   },
  getGenerators   otherGenerators: [],
  getGenerators   schemaPath: '/Users/alberto/Code/prisma-215-new-dev-flow/starter/prisma/schema.prisma',
  getGenerators   version: '94e94e327ccbd6d374c9a079ed59f89b077348e4',
  getGenerators   binaryPaths: {
  getGenerators     queryEngine: {
  getGenerators       darwin: '/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/cli/engines/a05654aa8247646280785347477a47b2bbf13b91/query-engine-darwin'
  getGenerators     }
  getGenerators   }
Running generate... - Prisma Client
__dirname /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build
/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build
  GeneratorProcess 2021-01-19T13:06:50.257Z loadEnv Error: Cannot find module '/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/.prisma/package.json' +0ms
  GeneratorProcess Require stack: +0ms
  GeneratorProcess - /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js +1ms
  GeneratorProcess     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15) +0ms
  GeneratorProcess     at Function.Module._load (internal/modules/cjs/loader.js:923:27) +0ms
  GeneratorProcess     at Module.require (internal/modules/cjs/loader.js:1140:19) +0ms
  GeneratorProcess     at require (internal/modules/cjs/helpers.js:75:18) +0ms
  GeneratorProcess     at /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72243:24 +0ms
  GeneratorProcess     at runMatcher (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:71968:25) +0ms
  GeneratorProcess     at AsyncFunction.module2.exports.sync (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:71975:25) +0ms
  GeneratorProcess     at getProjectRootEnvPath (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72239:43) +0ms
  GeneratorProcess     at Object.getEnvPaths2 (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72218:31) +0ms
  GeneratorProcess     at TSClient.toJS (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:77817:34) +0ms
  GeneratorProcess 2021-01-19T13:06:50.257Z loadEnv Error: Cannot find module '/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/package.json' +0ms
  GeneratorProcess Require stack: +0ms
  GeneratorProcess - /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js +0ms
  GeneratorProcess     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1080:15) +0ms
  GeneratorProcess     at Function.Module._load (internal/modules/cjs/loader.js:923:27) +0ms
  GeneratorProcess     at Module.require (internal/modules/cjs/loader.js:1140:19) +0ms
  GeneratorProcess     at require (internal/modules/cjs/helpers.js:75:18) +0ms
  GeneratorProcess     at /Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72243:24 +0ms
  GeneratorProcess     at runMatcher (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:71968:25) +0ms
  GeneratorProcess     at AsyncFunction.module2.exports.sync (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:71975:25) +0ms
  GeneratorProcess     at getProjectRootEnvPath (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72239:43) +0ms
  GeneratorProcess     at Object.getEnvPaths2 (/Users/alberto/Code/prisma-215-new-dev-flow/starter/node_modules/@prisma/client/generator-build/index.js:72218:31) +1ms
✔ Generated Prisma Client (2.15.0-dev.65) to ./node_modules/@prisma/client in 82ms

Everything is now in sync.
```
## Environment & setup

Tested with `2.15.0-dev.65`, using TypeScript and MySQL.

```

```

--- a/src/packages/migrate/src/__tests__/DbSeed.test.ts
+++ b/src/packages/migrate/src/__tests__/DbSeed.test.ts
@@ -76,8 +76,8 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                                                                                            🌱  Your database has been seeded.
-                                                                                                                                                                                                                                                          `)
+                                                                                                                                                                                                                                                                                                                        🌱  Your database has been seeded.
+                                                                                                                                                                                                                                                                    `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`node seed.js\` ...`)
@@ -96,8 +96,8 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                                                                                                🌱  Your database has been seeded.
-                                                                                `)
+                                                                                                            🌱  Your database has been seeded.
+                                                                                          `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`ts-node seed.ts\` ...`)
@@ -116,8 +116,8 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                                                                                            🌱  Your database has been seeded.
-                                                                                                                                                                                                                                                          `)
+                                                                                                                                                                                                                                                                                                                        🌱  Your database has been seeded.
+                                                                                                                                                                                                                                                                    `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`sh seed.sh\` ...`)
@@ -140,7 +140,7 @@ describe('seed', () => {
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`node seed.js\` ...`)
     expect(ctx.mocked['console.error'].mock.calls.join()).toMatchInlineSnapshot(
-      'Command failed with exit code 1',
+      ``,
     )
   })
 })

--- a/src/packages/migrate/src/__tests__/DbSeed.test.ts
+++ b/src/packages/migrate/src/__tests__/DbSeed.test.ts
@@ -1,4 +1,4 @@
-import execa from 'execa'
+import fs from 'fs-jetpack'
 import { DbSeed } from '../commands/DbSeed'
 import { consoleContext, Context } from './__helpers__/context'
 
@@ -22,7 +22,7 @@ describe('seed', () => {
 
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).rejects.toMatchInlineSnapshot(`
-            More than one seed file was found in \`prisma\`.
+            More than one seed file was found in \`prisma\` directory.
             This command only supports one seed file: Use \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\`.
           `)
     expect(
@@ -40,7 +40,7 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).rejects.toMatchInlineSnapshot(`
             No seed file found.
-            Create a \`seed.ts\` or \`.js\` or \`.sh\` or \`.go\` file in the prisma folder.
+            Create a \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\` file in the prisma directory.
           `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
@@ -55,7 +55,7 @@ describe('seed', () => {
 
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).rejects.toMatchInlineSnapshot(`
-            More than one seed file was found in \`prisma\`.
+            More than one seed file was found in \`prisma\` directory.
             This command only supports one seed file: Use \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\`.
           `)
     expect(
@@ -76,8 +76,8 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                    🌱  Your database has been seeded.
-                                                                                                                                                                                              `)
+                                                                                                                                                                                                                                                                                                            🌱  Your database has been seeded.
+                                                                                                                                                                                                                                                          `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`node seed.js\` ...`)
@@ -96,8 +96,8 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                        🌱  Your database has been seeded.
-                    `)
+                                                                                                🌱  Your database has been seeded.
+                                                                                `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`ts-node seed.ts\` ...`)
@@ -116,13 +116,32 @@ describe('seed', () => {
     const result = DbSeed.new().parse(['--preview-feature'])
     await expect(result).resolves.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                    🌱  Your database has been seeded.
-                                                                                                                                                                                              `)
+                                                                                                                                                                                                                                                                                                            🌱  Your database has been seeded.
+                                                                                                                                                                                                                                                          `)
     expect(
       ctx.mocked['console.info'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(`Running \`sh seed.sh\` ...`)
     expect(
       ctx.mocked['console.error'].mock.calls.join('\n'),
     ).toMatchInlineSnapshot(``)
+  })
+
+  it('one broken seed.js file', async () => {
+    ctx.fixture('edited-and-draft')
+    ctx.fs.remove('prisma/seed.js')
+    ctx.fs.remove('prisma/seed.ts')
+    ctx.fs.remove('prisma/seed.sh')
+    ctx.fs.remove('prisma/seed.go')
+    fs.write('prisma/seed.js', 'BROKENCODE;;;;;')
+
+    const result = DbSeed.new().parse(['--preview-feature'])
+    await expect(result).rejects.toThrowError()
+    expect(
+      ctx.mocked['console.info'].mock.calls.join('\n'),
+    ).toMatchInlineSnapshot(`Running \`node seed.js\` ...`)
+    expect(ctx.mocked['console.error'].mock.calls.join()).toMatchInlineSnapshot(
+      'Command failed with exit code 1',
+      ``,
+    )
   })
 })

--- a/src/packages/migrate/src/__tests__/DbSeed.test.ts
+++ b/src/packages/migrate/src/__tests__/DbSeed.test.ts
@@ -141,7 +141,6 @@ describe('seed', () => {
     ).toMatchInlineSnapshot(`Running \`node seed.js\` ...`)
     expect(ctx.mocked['console.error'].mock.calls.join()).toMatchInlineSnapshot(
       'Command failed with exit code 1',
-      ``,
     )
   })
 })

--- a/src/packages/migrate/src/__tests__/MigrateDev.test.ts
+++ b/src/packages/migrate/src/__tests__/MigrateDev.test.ts
@@ -361,6 +361,7 @@ describe('sqlite', () => {
       Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
 
 
+
       The following migration(s) have been created and applied from new schema changes:
 
       migrations/
@@ -409,6 +410,7 @@ describe('sqlite', () => {
       Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
 
 
+
       The following migration(s) have been applied:
 
       migrations/
@@ -437,6 +439,7 @@ describe('sqlite', () => {
       .toMatchInlineSnapshot(`
       Prisma schema loaded from prisma/schema.prisma
       Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
+
 
 
       The following migration(s) have been applied:
@@ -595,11 +598,11 @@ describe('sqlite', () => {
 
     await expect(result).rejects.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ⚠️ We found changes that cannot be executed:
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ⚠️ We found changes that cannot be executed:
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      • Step 0 Made the column \`fullname\` on table \`Blog\` required, but there are 1 existing NULL values.
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  • Step 0 Made the column \`fullname\` on table \`Blog\` required, but there are 1 existing NULL values.
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              `)
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        `)
     expect(ctx.mocked['console.info'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
       Prisma schema loaded from prisma/schema.prisma
@@ -658,10 +661,10 @@ describe('sqlite', () => {
     expect(ctx.mocked['console.log'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
 
-                                                                                                                                                ⚠️  There will be data loss when applying the migration:
+                                                                                                                                                      ⚠️  There will be data loss when applying the migration:
 
-                                                                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
-                                                                                                `)
+                                                                                                                                                        • You are about to drop the \`Blog\` table, which is not empty (2 rows).
+                                                                                                    `)
     expect(ctx.mocked['console.error'].mock.calls).toMatchSnapshot()
   })
 
@@ -682,10 +685,10 @@ describe('sqlite', () => {
     expect(ctx.mocked['console.log'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
 
-                                                                                                                                                ⚠️  There will be data loss when applying the migration:
+                                                                                                                                                      ⚠️  There will be data loss when applying the migration:
 
-                                                                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
-                                                                                                `)
+                                                                                                                                                        • You are about to drop the \`Blog\` table, which is not empty (2 rows).
+                                                                                                    `)
     expect(ctx.mocked['console.error'].mock.calls).toMatchSnapshot()
   })
 
@@ -731,6 +734,7 @@ describe('sqlite', () => {
 
 
       Running \`node seed.js\` ...
+
       The following migration(s) have been applied:
 
       migrations/
@@ -762,6 +766,7 @@ describe('sqlite', () => {
 
 
       Running \`node seed.js\` ...
+
       The following migration(s) have been applied:
 
       migrations/
@@ -793,6 +798,7 @@ describe('sqlite', () => {
       .toMatchInlineSnapshot(`
       Prisma schema loaded from prisma/schema.prisma
       Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
+
 
 
       The following migration(s) have been applied:

--- a/src/packages/migrate/src/__tests__/MigrateDev.test.ts
+++ b/src/packages/migrate/src/__tests__/MigrateDev.test.ts
@@ -595,11 +595,11 @@ describe('sqlite', () => {
 
     await expect(result).rejects.toMatchInlineSnapshot(`
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ⚠️ We found changes that cannot be executed:
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ⚠️ We found changes that cannot be executed:
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      • Step 0 Made the column \`fullname\` on table \`Blog\` required, but there are 1 existing NULL values.
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      • Step 0 Made the column \`fullname\` on table \`Blog\` required, but there are 1 existing NULL values.
 
-                                                                                                                                                                                                                                                                                                                                                                                                                                              `)
+                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              `)
     expect(ctx.mocked['console.info'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
       Prisma schema loaded from prisma/schema.prisma
@@ -658,10 +658,10 @@ describe('sqlite', () => {
     expect(ctx.mocked['console.log'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
 
-                                                                                                ⚠️  There will be data loss when applying the migration:
+                                                                                                                                                ⚠️  There will be data loss when applying the migration:
 
-                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
-                                                                `)
+                                                                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
+                                                                                                `)
     expect(ctx.mocked['console.error'].mock.calls).toMatchSnapshot()
   })
 
@@ -682,10 +682,10 @@ describe('sqlite', () => {
     expect(ctx.mocked['console.log'].mock.calls.join('\n'))
       .toMatchInlineSnapshot(`
 
-                                                                                                ⚠️  There will be data loss when applying the migration:
+                                                                                                                                                ⚠️  There will be data loss when applying the migration:
 
-                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
-                                                                `)
+                                                                                                                                                  • You are about to drop the \`Blog\` table, which is not empty (2 rows).
+                                                                                                `)
     expect(ctx.mocked['console.error'].mock.calls).toMatchSnapshot()
   })
 
@@ -711,6 +711,105 @@ describe('sqlite', () => {
     ).toMatchInlineSnapshot(
       `Response: Datasource provider arrays are no longer supported in migrate. Please change your datasource to use a single provider. Read more at https://pris.ly/multi-provider-deprecation`,
     )
+  })
+
+  it('one seed file', async () => {
+    ctx.fixture('edited-and-draft')
+    fs.write('prisma/seed.js', 'console.log("Hello from generated seed")')
+
+    prompt.inject(['y'])
+
+    const result = MigrateDev.new().parse(['--preview-feature'])
+
+    await expect(result).resolves.toMatchInlineSnapshot(
+      `Everything is now in sync.`,
+    )
+    expect(ctx.mocked['console.info'].mock.calls.join('\n'))
+      .toMatchInlineSnapshot(`
+      Prisma schema loaded from prisma/schema.prisma
+      Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
+
+
+      Running \`node seed.js\` ...
+      The following migration(s) have been applied:
+
+      migrations/
+        └─ 20201231000000_test/
+          └─ migration.sql
+        └─ 20201231000000_draft/
+          └─ migration.sql
+
+    `)
+    expect(ctx.mocked['console.log'].mock.calls.join()).toMatchSnapshot()
+    expect(ctx.mocked['console.error'].mock.calls.join()).toMatchSnapshot()
+  })
+
+  it('one broken seed file', async () => {
+    ctx.fixture('edited-and-draft')
+    fs.write('prisma/seed.js', 'BROKENCODE;;;;;')
+
+    prompt.inject(['y'])
+
+    const result = MigrateDev.new().parse(['--preview-feature'])
+
+    await expect(result).resolves.toMatchInlineSnapshot(
+      `Everything is now in sync.`,
+    )
+    expect(ctx.mocked['console.info'].mock.calls.join('\n'))
+      .toMatchInlineSnapshot(`
+      Prisma schema loaded from prisma/schema.prisma
+      Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
+
+
+      Running \`node seed.js\` ...
+      The following migration(s) have been applied:
+
+      migrations/
+        └─ 20201231000000_test/
+          └─ migration.sql
+        └─ 20201231000000_draft/
+          └─ migration.sql
+
+    `)
+    expect(ctx.mocked['console.log'].mock.calls.join()).toMatchSnapshot()
+    expect(ctx.mocked['console.error'].mock.calls.join()).toContain(
+      'Command failed with exit code 1',
+    )
+  })
+
+  it('multple seed files', async () => {
+    ctx.fixture('edited-and-draft')
+    fs.write('prisma/seed.sh', 'echo "Hello from generated seed"')
+    fs.write('prisma/seed.js', 'console.log("Hello from generated seed")')
+
+    prompt.inject(['y'])
+
+    const result = MigrateDev.new().parse(['--preview-feature'])
+
+    await expect(result).resolves.toMatchInlineSnapshot(
+      `Everything is now in sync.`,
+    )
+    expect(ctx.mocked['console.info'].mock.calls.join('\n'))
+      .toMatchInlineSnapshot(`
+      Prisma schema loaded from prisma/schema.prisma
+      Datasource "my_db": SQLite database "dev.db" at "file:dev.db"
+
+
+      The following migration(s) have been applied:
+
+      migrations/
+        └─ 20201231000000_test/
+          └─ migration.sql
+        └─ 20201231000000_draft/
+          └─ migration.sql
+
+    `)
+    expect(ctx.mocked['console.log'].mock.calls.join()).toMatchSnapshot()
+    expect(ctx.mocked['console.error'].mock.calls.join())
+      .toMatchInlineSnapshot(`
+      Error: More than one seed file was found in \`prisma\` directory.
+      This command only supports one seed file: Use \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\`.
+    `)
   })
 })
 

--- a/src/packages/migrate/src/__tests__/MigrateReset.test.ts
+++ b/src/packages/migrate/src/__tests__/MigrateReset.test.ts
@@ -208,7 +208,7 @@ describe('reset', () => {
 
     const result = MigrateReset.new().parse(['--preview-feature'])
     await expect(result).rejects.toMatchInlineSnapshot(`
-            More than one seed file was found in \`prisma\`.
+            More than one seed file was found in \`prisma\` directory.
             This command only supports one seed file: Use \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\`.
           `)
     expect(

--- a/src/packages/migrate/src/__tests__/fixtures/nativeTypes-mssql/prisma/schema.prisma
+++ b/src/packages/migrate/src/__tests__/fixtures/nativeTypes-mssql/prisma/schema.prisma
@@ -1,0 +1,15 @@
+datasource db {
+  provider = "sqlserver"
+  url      = env("TEST_MSSQL_JDBC_URI")
+}
+
+generator client {
+  previewFeatures = ["microsoftSqlServer", "nativeTypes"]
+  provider = "prisma-client-js"
+}
+
+model User {
+  id   Int     @id @default(autoincrement())
+  name String?
+  test BigInt
+}

--- a/src/packages/migrate/src/commands/MigrateDev.ts
+++ b/src/packages/migrate/src/commands/MigrateDev.ts
@@ -30,6 +30,7 @@ import {
 import { getMigrationName } from '../utils/promptForMigrationName'
 import { throwUpgradeErrorIfOldMigrate } from '../utils/detectOldMigrate'
 import { printDatasource } from '../utils/printDatasource'
+import { tryToRunSeed, detectSeedFiles } from '../utils/seed'
 
 const debug = Debug('migrate:dev')
 
@@ -171,6 +172,17 @@ ${chalk.bold('Examples')}
 
       // Do the reset
       await migrate.reset()
+
+      // Run seed if 1 or more seed files are present
+      // And catch the error to continue execution
+      try {
+        const detected = detectSeedFiles()
+        if (detected.numberOfSeedFiles > 0) {
+          await tryToRunSeed()
+        }
+      } catch (e) {
+        console.error(e)
+      }
     }
 
     const { appliedMigrationNames } = await migrate.applyMigrations()

--- a/src/packages/migrate/src/commands/MigrateDev.ts
+++ b/src/packages/migrate/src/commands/MigrateDev.ts
@@ -182,6 +182,8 @@ ${chalk.bold('Examples')}
         }
       } catch (e) {
         console.error(e)
+      } finally {
+        console.info() // empty line
       }
     }
 

--- a/src/packages/migrate/src/utils/seed.ts
+++ b/src/packages/migrate/src/utils/seed.ts
@@ -32,13 +32,13 @@ export async function tryToRunSeed() {
 
   if (detected.numberOfSeedFiles === 0) {
     throw new Error(`No seed file found.
-Create a \`seed.ts\` or \`.js\` or \`.sh\` or \`.go\` file in the prisma folder.`)
+Create a \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\` file in the prisma directory.`)
   } else if (detected.numberOfSeedFiles > 1) {
     throw new Error(
       `More than one seed file was found in \`${path.relative(
         process.cwd(),
         path.dirname(detected.seedPath),
-      )}\`.
+      )}\` directory.
 This command only supports one seed file: Use \`seed.ts\`, \`.js\`, \`.sh\` or \`.go\`.`,
     )
   } else {

